toolz/itertoolz.py

```diff
@@ -496,6 +496,34 @@ def concatv(*seqs):
     return concat(seqs)
 
 
+def duplicate(seq, n=1):
+    """ Gets each element multiple times.
+
+    Like ``itertools.repeat`` this will repeat each element n-times.
+    However, it will do this for each element of the sequence.
+
+     Args:
+
+         seq(iterable):           The sequence to grab items from.
+
+         n(integral):             Number of repeats for each element.
+
+     Returns:
+
+         generator:               A generator of repeated elements.
+
+     Examples:
+
+         >>> list(duplicate([1, 2, 3], 2))
+         [1, 1, 2, 2, 3, 3]
+    """
+
+    assert (n >= 0), "n must be positive, but got n = " + repr(n)
+    assert ((n % 1) == 0), "n must be an integer, but got n = " + repr(n)
+
+    return concat(map(lambda _: itertools.repeat(_, n), seq))
```

Comment:
for a toolzier implementation, you could replace the lambda with `flip(itertools.repeat, n)`